File: ysKim/week1/2018KAKAO-1.py
def solution(dartResult):
    answer = 0
    result = []
    ind = 0
    for i in range(len(dartResult)):
        if dartResult[i].isdigit():
            if len(result) > 0 and int(result[len(result) - 1]) == 10 and dartResult[i] == '0':
                continue
            if dartResult[i + 1].isdigit():
                result.append(10)
                ind = len(result) - 1
                continue
            result.append(dartResult[i])
            ind = len(result) - 1
        elif dartResult[i].isalpha():
            print(dartResult[i])
            if dartResult[i] == 'D':
                result[ind] = int(result[ind]) ** 2
            elif dartResult[i] == 'T':
                result[ind] = int(result[ind]) ** 3
        else:
            if dartResult[i] == '*':
                if ind == 0:
                    result[ind] = int(result[ind]) * 2
                else:
                    result[ind - 1] = int(result[ind - 1]) * 2
                    result[ind] = int(result[ind]) * 2
            else:
                result[ind] = int(result[ind]) * (-1)
    print(result)
    for j in range(len(result)):
        answer = answer + int(result[j])
    return answer
if __name__ == '__main__':
    print(solution('10T10S1T*'))

File: jhYang/week1/2018KAKAO-1.cpp
// 다트게임
#include <iostream>

using namespace std;

bool isDigit(int c){
    if(c>='0'&&c<='9')
        return true;
    return false;
}

int solution(string dartResult) {
    int answer = 0;
    bool f = false;
    int temp = -1;
    int ptmp = -100000;
    for(int i = 0; i<dartResult.size(); i++){
        int dr = dartResult[i];
        if(isDigit(dr)){
            temp = int(dr)-'0';
            if(dr=='1'&&isDigit(int(dartResult[i+1]))){
                temp = 10;
                i++;
            }
            continue;
        }
        if(dr == 'S' || dr == 'D' || dr == 'T'){
            if(dr=='S') {}
            if(dr=='D') {temp*=temp;}
            if(dr=='T') {temp*=temp*temp;}
            if(i+1<dartResult.size() && !isDigit(int(dartResult[i+1])))
                dr=dartResult[++i];
        }
        if(dr=='#') {temp*=-1;}
        if(dr=='*') {ptmp*=2; temp*=2;}
        if(ptmp <= -100000){
            ptmp = temp;
            temp = -1;
        }
        else{
            answer+=ptmp;
            ptmp = temp;
            temp = -1;
        }
    }
    answer+=ptmp;
    return answer;
}
int main(){
    string dartResult;
    cin>>dartResult;
    cout<<solution(dartResult);
}